src/g/selection/SelectionCol.js

```diff
@@ -200,34 +225,328 @@ const SelectionManager = Collection.extend({
+      const { id } = model.attributes
+      acc[idx] = id
+      return acc
+    }, {});
+
+    const lastSelection = this.lastSelection;
+
+    if (!lastSelection || !this._isSelectionValid(lastSelection)) {
+      this.add(this._getSelsWithLabelsForRows([selection]), {silent: true});
+      return;
+    }
+
+    const lastSelectionType = lastSelection.get("type");
+    const lSelSeqIdIdx = seqIdToIdxMap[lastSelection.get("seqId")];
+    const lSelXStart = lastSelection.get("xStart");
+    const lSelXEnd = lastSelection.get("xEnd");
+
+    const selectionType = selection.get("type");
+    const selSeqIdIdx = seqIdToIdxMap[selection.get("seqId")];
+    const selXStart = selection.get("xStart");
+    const selXEnd = selection.get("xEnd");
+
+    const minXStart = Math.min(lSelXStart, selXStart);
+    const maxXEnd = Math.max(lSelXEnd, selXEnd);
+    const minSeqIdIdx = Math.min(lSelSeqIdIdx, selSeqIdIdx);
+    const maxSeqIdIdx = Math.max(lSelSeqIdIdx, selSeqIdIdx);
+
+    if (lastSelectionType === "row" && selectionType === "row") {
+      // Select all rows between the last selection and the current selection
+      const selections = [];
+      for (let i = minSeqIdIdx; i <= maxSeqIdIdx; i++) {
+        selections.push(new rowsel({seqId: idxToSeqIdMap[i]}));
+      }
+      this.add(this._getSelsWithLabelsForRows(selections), {silent: true});
+    } else if (lastSelectionType === "column" && selectionType === "column") {
+      // Select all columns between the last selection and the current selection
+      const columns = [];
+      for (let i = minXStart; i <= maxXEnd; i++) {
+          columns.push(new columnsel({xStart: i, xEnd: i}));
+      }
+      this.add(columns, {silent: true});
+    } else if (lastSelectionType === "pos" && selectionType === "pos" ) {
+      // Select all residues between the last selection and the current selection
+      const positions = [];
+      for (let i = minSeqIdIdx; i <= maxSeqIdIdx; i++) {
+        for (let j = minXStart; j <= maxXEnd; j++) {
+          positions.push(new possel({xStart: j, xEnd: j, seqId: idxToSeqIdMap[i]}));
+        }
+      }
+      this.add(positions, {silent: true});
+    } else if (lastSelectionType === "label" && selectionType === "label" ) {
+      // Select all labels between the last selection and the current selection
+      const labels = [];
+      for (let i = minSeqIdIdx; i <= maxSeqIdIdx; i++) {
+        labels.push(new labelsel({seqId: idxToSeqIdMap[i]}));
+      }
+      this.add(labels, {silent: true});
+    } else if (lastSelectionType === "row" && selectionType === "pos") {
+      const positions = [];
+      for (let i = minSeqIdIdx; i <= maxSeqIdIdx; i++) {
+        for (let j = selXStart; j <= selXEnd; j++) {
+          positions.push(new possel({xStart: j, xEnd: j, seqId: idxToSeqIdMap[i]}));
         }
       }
+      this.add(positions, {silent: true});
+    } else if (lastSelectionType === "column" && selectionType === "pos") {
```

Comment:
I included this case, taking inspiration from LD's grid.

Reply:
seems like an over kill but since the work is already done, let's keep it.
Please add a NOTE explaining what this case is for though.
And don't we need the row equivalent to this?